front/components/assistant/conversation/BlockedActionsProvider.tsx

```diff
@@ -417,173 +194,104 @@ export function BlockedActionsProvider({
     });
 
     if (!result.success) {
+      setErrorMessage("Failed to assess action approval. Please try again.");
       return;
     }
 
-    removeCompletedAction(blockedAction.actionId);
-
     setSubmitStatus(null);
     setNeverAskAgain(false);
     setErrorMessage(null);
 
-    setValidatedActions((c) => c + 1);
+    // Move to next validation or close dialog if done
+    if (currentValidationIndex + 1 < pendingValidations.length) {
+      setCurrentValidationIndex(currentValidationIndex + 1);
+    } else {
+      pendingValidations.forEach((item) => {
+        removeCompletedAction(item.blockedAction.actionId);
+      });
```

Comment:
Let's add a small comment explaining why we're not popping the elements once processed